src/time.rs

```diff
@@ -0,0 +1,170 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Instant {
+    millis: u64,
```

Comment:
On second thought, I think this should be an `i64`. Then we do not need `checked_sub` and `saturating_sub` at all! The zero instant would just be an arbitrary point in a (nearly) infinite timeline.

Reply:
What would a negative value indicate?

Reply:
"Before startup", since 0 is usually "startup".